.github/workflows/cd.yml

```diff
@@ -18,29 +18,34 @@ jobs:
       profile: "debug"
 
   # Builds debug images with release binaries for compatibility tests in case the merge_group was skipped.
-  build-data-plane-test-images:
+  data-plane-test-images:
     uses: ./.github/workflows/_data-plane.yml
     secrets: inherit
     with:
       image_prefix: "debug"
       stage: "debug"
       profile: "release"
 
-  build-control-plane-images:
-    uses: ./.github/workflows/_control-plane.yml
-    secrets: inherit
-
-  # Re-run CI checks to make sure everything's green, since "Merging as administrator"
-  # won't trigger these in the merge group.
-  ci:
-    uses: ./.github/workflows/ci.yml
+  # Build data-plane prod images for staging and production environments.
+  data-plane-prod-images:
+    uses: ./.github/workflows/_data-plane.yml
     secrets: inherit
     with:
       profile: "release"
       stage: "release"
 
+  # Build control-plane prod images for staging and production environments.
+  control-plane-images:
+    uses: ./.github/workflows/_control-plane.yml
+    secrets: inherit
+
+  # Build loadtest binaries for QA environment.
+  loadtest-binaries:
+    uses: ./.github/workflows/_loadtest.yml
+    secrets: inherit
+
   notify:
```

Comment:
We could split this `notify` job into several ones and kick off different pipelines in the `infra` repository for staging and QA.